src/pages/api/articles/[id].tsx

```diff
@@ -0,0 +1,12 @@
+import { articles } from "../../../../data";
+
+export default ({ query: { id } }, res) => {
+  const filtered = articles.filter((a) => a.id === id);
```

Comment:
一つ見つければいいから find でよくない？
```suggestion
  const articleFound = articles.find((a) => a.id === id); // Article | undefined
```